src/plugin/adapter-registry/adapter-capabilities.ts

```diff
@@ -122,15 +128,24 @@ export class AdapterVariableTracker implements vscode.DebugAdapterTracker {
         return candidate.presentationHint !== 'registers' && candidate.name !== 'Registers';
     }
 
-    protected variableToVariableRange(_variable: DebugProtocol.Variable, _session: vscode.DebugSession): Promise<VariableRange | undefined> {
+    protected variableToVariableRange(_variable: DebugProtocol.Variable, _session: vscode.DebugSession, _context?: Context): Promise<VariableRange | undefined> {
         throw new Error('To be implemented by derived classes!');
     }
 
     /** Resolves the address of a given variable in bytes within the current context. */
-    getAddressOfVariable?(variableName: string, session: vscode.DebugSession): Promise<string | undefined>;
+    getAddressOfVariable?(variableName: string, session: vscode.DebugSession, context?: Context): Promise<string | undefined>;
 
     /** Resolves the size of a given variable in bytes within the current context. */
-    getSizeOfVariable?(variableName: string, session: vscode.DebugSession): Promise<bigint | undefined>;
+    getSizeOfVariable?(variableName: string, session: vscode.DebugSession, context?: Context): Promise<bigint | undefined>;
+
+    readMemory?(session: vscode.DebugSession, params: ReadMemoryArguments, context: Context): Promise<ReadMemoryResult>;
+
+    writeMemory?(session: vscode.DebugSession, params: WriteMemoryArguments, context: Context): Promise<WriteMemoryResult>;
```

Comment:
In other cases, the context is optional; why is it required here?

Reply:
Whoops. Good catch. Thanks!